src/mlpack/tests/loss_functions_test.cpp

```diff
@@ -436,4 +437,42 @@ BOOST_AUTO_TEST_CASE(SimpleMeanBiasErrorTest)
   BOOST_REQUIRE_EQUAL(output.n_elem, 1);
 }
 
+/*
+ * Simple tests for the Hinge Embedding Loss Function.
+ */
+BOOST_AUTO_TEST_CASE(HingeEmbeddingLossTest)
+{
+  arma::mat input, target, output;
+  double loss;
+  HingeEmbeddingLoss<> module;
+
+  // Test the Forward Function. Loss should be 0 if input = target.
+  input = arma::ones(10, 1);
+  target = arma::ones(10, 1);
+  loss = module.Forward(std::move(input), std::move(target));
+  BOOST_REQUIRE_EQUAL(loss, 0);
+
+  // Test the Backward function for input = target.
+  module.Backward(std::move(input), std::move(target), std::move(output));
+  for (double el : output)
+  {
+    // For input = target we should get 0.0 everywhere.
+    BOOST_REQUIRE_CLOSE(el, 0.0, 1e-5);
+  }
+
+  BOOST_REQUIRE_EQUAL(output.n_rows, input.n_rows);
+  BOOST_REQUIRE_EQUAL(output.n_cols, input.n_cols);
+
+  // Test the Forward function. Loss should be 0.879
```

Comment:
Could you also add a stop here. Thanks for all your work.

Reply:
Taken care of in this commit.
Thanks for noticing.